cli/src/main/java/org/jboss/as/cli/embedded/EmbedServerHandler.java

```diff
@@ -224,13 +225,19 @@ protected void doHandle(CommandContext ctx) throws CommandLineException {
 
             String[] cmds = cmdsList.toArray(new String[cmdsList.size()]);
 
-            EmbeddedManagedProcess server;
+            final Configuration.Builder configBuilder;
             if (this.jbossHome == null) {
                 // Modular environment
-                server = EmbeddedProcessFactory.createStandaloneServer(ModuleLoader.forClass(getClass()), jbossHome, cmds);
+                configBuilder = Configuration.Builder.of(jbossHome)
```

Comment:
this piece of code is a bit confusing.
`this.jbossHome` which is a `ArgumentWithValue` can be `null` but `jbossHome` which is `File` can not be `null`  (according to the `getJBossHome` method).

The presence of a `this.jbossHome` means we are in a modular environment and then use the `jbossHome` value that comes from the env property `JBOSS_HOME`.
Else we are not in an environment module and uses the absolute path from the `--jboss-home` argument.

Reply:
Correct and I do agree it's a bit confusing. I'll add a comment as to how it works. I just copied what the previous version used for it's home directory.